### PR TITLE
Move organization delete button into org details

### PR DESCRIPTION
**Pull request by axbarsan**
Closes https://github.com/giantswarm/giantswarm/issues/13914

This PR includes:
* A bit of clean up in the organization components
* Removing the organization delete button from the organization list view
* Adding a delete button to the organization detail view
	* Disabling the delete button and explaining that organizations can't be deleted when they contain clusters
	* Disabling the delete button while credentials are loading
	* Disabling the delete button and explaining that organizations can't be deleted when they have credentials

### Preview

<details>
<summary>Organization ready to be deleted</summary>

image

</details>

<details>
<summary>Organization has clusters, can't be deleted</summary>

image

</details>

<details>
<summary>Organization is loading credentials, can't be deleted</summary>

image

</details>

<details>
<summary>Organization has credentials, can't be deleted</summary>

image

</details>

After clicking on the `Delete organization` button, the same modal that we had in the previous implementation is opened, and if the deletion succeedes, we get redirected to the organization list view.

### CODE REVIEW

**Review by marians — APPROVED** [review]
LGTM! (didn't look into the code)

File: src/components/UI/OrganizationList/OrganizationList.js [hunk 1]
```diff
@@ -19,13 +19,11 @@ const OrganizationList = ({ provider, ...props }) => {
           <StyledTableHeader centered={true}>Clusters</StyledTableHeader>
           <StyledTableHeader centered={true}>Members</StyledTableHeader>
 
-          {provider !== Providers.KVM && (
+          {(provider === Providers.AWS || provider === Providers.AZURE) && (
```
Comment (oponder) [inline src/components/UI/OrganizationList/OrganizationList.js:22]: This piece of business logic is duplicated in `src/components/Organizations/Detail/View.js`

and `src/components/UI/OrganizationList/OrganizationListRow.js`

I realise thats not something you introduced. But if you feel like extracting it to a common place, that would be 💯

**Review by oponder — APPROVED** [review]
LGTM! Just noticed a business logic extraction opportunity.

File: src/components/UI/OrganizationList/OrganizationList.js (hunk 1 above)
Comment (axbarsan) [reply src/components/UI/OrganizationList/OrganizationList.js]: Very good idea 👍🏻

**Pull request merged**